core/Inst.hpp

```diff
@@ -324,6 +336,24 @@ namespace olympia
 
         const OpInfoList & getDestOpInfoList() const { return opcode_info_->getDestOpInfoList(); }
 
+        bool hasZeroRegSource() const
+        {
+            return std::any_of(getSourceOpInfoList().begin(), getSourceOpInfoList().end(),
+                [](const mavis::OperandInfo::Element & elem)
+                {
+                    return elem.field_value == 0;
+                });
+        }
+
+        bool hasZeroRegDest() const
+        {
+            return std::any_of(getDestOpInfoList().begin(), getDestOpInfoList().end(),
+                [](const mavis::OperandInfo::Element & elem)
+                {
+                    return elem.field_value == 0;
+                });
+        }
```

Comment:
Wonder if this could be part of the static information on the instruction?

Perhaps it doesn't matter...